### ISSUE TITLE
function.js does not exist in typescript project

### ISSUE THREAD

**Issue by trigonometric-act** [google-github-actions/deploy-cloud-functions#366]
### TL;DR

Can't deploy cloud functions with github action, I get the following error: google-github-actions/deploy-cloud-functions failed with: operation failed: Build failed: function.js does not exist; Error ID: 7485c5b6

### Expected behavior

Expecting to be able to deploy.

### Observed behavior

From my understanding the job is trying to deploy the function from a js file, but that file does not exists in the moment of deploy because .gitignore ignores js files in the ts version of cloud functions, as there is no documentation for typescript projects I suppose that doesn't matter.
also my package json has the following entry point: 
"main": "lib/index.js"
Log when executes deploy job:
```
Run google-github-actions/deploy-cloud-functions@v1
Created zip file from './' at '/tmp/cfsrc-80f55d8efb44d18e6ea48fad.zip'
Creating new Cloud Function revision
Deploying Cloud Function
Still deploying Cloud Function (1/n)
Still deploying Cloud Function (2/n)
Still deploying Cloud Function (3/n)
Still deploying Cloud Function (4/n)
Still deploying Cloud Function (5/n)
Still deploying Cloud Function (6/n)
Still deploying Cloud Function (7/n)
Error: google-github-actions/deploy-cloud-functions failed with: operation failed: Build failed: function.js does not exist; Error ID: 7485c5b6
```

### Action YAML

```yaml
# This is a basic workflow to help you get started with Actions
name: CD

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the main branch
  push:
    branches: [master]
  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "deploy"
  deploy:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Add "id-token" with the intended permissions.
    permissions:
      contents: "read"
      id-token: "write"

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v1
      - id: "auth"
        name: "Authenticate to Google Cloud"
        uses: "google-github-actions/auth@v1"
        with:
          workload_identity_provider: #####
          service_account: #####

      - id: "deploy"
        uses: "google-github-actions/deploy-cloud-functions@v1"
        with:
          # Name of the Cloud Function, same as the entry point name
          name: callGit
          entry_point: callGit
          runtime: nodejs16
          region: ####
      
      - name: make the url aunthenticated
        run: |
          gcloud functions add-iam-policy-binding callGit --member="allUsers" role="roles/cloudfunctions.invoker"  

      # Example of using the output
      - id: "test"
        run: 'curl "${{ steps.deploy.outputs.url }}"'
```

### Log output

```text
Current runner version: '2.299.1'
Operating System
Runner Image
Runner Image Provisioner
GITHUB_TOKEN Permissions
Secret source: Actions
Prepare workflow directory
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v1' (SHA:###)
Download action repository 'google-github-actions/auth@v1' (SHA: ###)
Download action repository 'google-github-actions/deploy-cloud-functions@v1' (SHA:###)
Run actions/checkout@v1
Syncing repository: #####
git version
git version 2.38.1
git lfs version
git-lfs/3.2.0 (GitHub; linux amd64; go 1.18.2)
git init "/home/runner/work/####
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint: 
hint: 	git config --global init.defaultBranch <name>
hint: 
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint: 
hint: 	git branch -m <name>
Initialized empty Git repository in /home/runner/work/###
git remote add origin https://github.com/####
git config gc.auto 0
git config --get-all http.https://github.com/###
git -c http.extraheader="AUTHORIZATION: basic ***" fetch --tags --prune --progress --no-recurse-submodules origin +refs/heads/*:refs/remotes/origin/*
Run google-github-actions/auth@v1
  with:
    workload_identity_provider: ####
    service_account: ####
    create_credentials_file: true
    export_environment_variables: true
    cleanup_credentials: true
    access_token_lifetime: 3600s
    access_token_scopes: https://www.googleapis.com/auth/cloud-platform
    retries: 0
    id_token_include_email: false
Created credentials file at "/home/runner/work/####
Run google-github-actions/deploy-cloud-functions@v1
Created zip file from './' at '/tmp/cfsrc-80f55d8efb44d18e6ea48fad.zip'
Creating new Cloud Function revision
Deploying Cloud Function
Still deploying Cloud Function (1/n)
Still deploying Cloud Function (2/n)
Still deploying Cloud Function (3/n)
Still deploying Cloud Function (4/n)
Still deploying Cloud Function (5/n)
Still deploying Cloud Function (6/n)
Still deploying Cloud Function (7/n)
Error: google-github-actions/deploy-cloud-functions failed with: operation failed: Build failed: function.js does not exist; Error ID: 7485c5b6
```

### Additional information

My index ts exports functions the following way:

export {callGit} from './one/path'
